zeebe/stream-platform/src/test/java/io/camunda/zeebe/stream/impl/StreamProcessorTest.java

```diff
@@ -1499,6 +1503,49 @@ void shouldUseProcessingFilter() {
     verify(streamPlatform.getDefaultMockedRecordProcessor(), never()).process(any(), any());
   }
 
+  @Test
+  void shouldShutDownStreamProcessorOnUncommittedStateException() throws Exception {
+    // given -- a transaction that fails unexpectedly, not with a regular `ZeebeDbException`.
+    streamPlatform.startStreamProcessor();
+    final var unexpectedException = new RuntimeException("Unexpected exception");
+
+    final var failingTransaction = mock(ZeebeDbTransaction.class);
+    doThrow(unexpectedException).when(failingTransaction).commit();
+
+    final var mockTransactionContext = mock(TransactionContext.class);
+    when(mockTransactionContext.getCurrentTransaction()).thenReturn(failingTransaction);
+
+    // given -- a modified processing state machine which uses the mock transaction context.
+    final var processingStateMachineField =
+        StreamProcessor.class.getDeclaredField("processingStateMachine");
+    processingStateMachineField.setAccessible(true);
+    final var processingStateMachine =
+        (ProcessingStateMachine)
+            processingStateMachineField.get(streamPlatform.getStreamProcessor());
+    final var transactionContextField =
+        ProcessingStateMachine.class.getDeclaredField("transactionContext");
+    transactionContextField.setAccessible(true);
+    transactionContextField.set(processingStateMachine, mockTransactionContext);
```

Comment:
🙈

Reply:
Couldn't we pass in a DB when creating the StreamProcessor, that fails on commit a transaction ? 🤔

Reply:
In theory sure :+1: But as I tried to justify in the PR description:
> It uses reflection to inject a broken transaction into the ProcessingStateMachine because it did not seem worth to add dedicated testing infrastructure for this test alone.

I don't see this being a common use case and hacking in support for this in `StreamPlatformExtension` and `StreamProcessorTestContext` did not seem worth the effort.

Reply:
I did this via https://github.com/camunda/camunda/pull/31459/commits/7651a1c119b3990f7a96846e02add83a0d241fdd (just to try it out and see how it looks).

I feel it can become useful if we want to test further edge cases regarding failing transaction or any db related. WDYT? 

Feel free to dismiss it if you disagree or dont like it :)